### PR TITLE
Master stock clean reporting hch

### PR DESCRIPTION
**Pull request by hch-odoo**
Task - https://www.odoo.com/web?debug=assets#id=36088&view_type=form&model=project.task&action=327&menu_id=4720
Pad - https://pad.odoo.com/p/r.e28516e06644727780892ff2cca4305d

### CODE REVIEW

**Comment by jir-odoo** [comment]
LGTM

**Comment by amoyaux** [comment]
Also it seems that framework js doesn't handle group_by with a list, I don't want if it's wanted or not but if the view return an error after rebase, you just need to remove '[' and ']' inside group by

File: addons/stock/report/report_stock_forecast.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
```
Comment (amoyaux) [inline addons/stock/report/report_stock_forecast.py]: remove this diff

**Comment by amoyaux** [comment]
@hch-odoo Thanks! A traceback is raise when you click on the list view from the pivot view. Could you check?

File: addons/stock/report/report_stock_forecast.py [hunk 2]
```diff
@@ -91,25 +96,59 @@ def init(self):
...
+                    res_list[line['product_id']].append(line['quantity'])  # Get cumulative quantity product wise
+                    line['cumulative_quantity'] = sum(res_list[line['product_id']])  # Sum of all quantities (i.e. cumulative quantity) product wise
+                    if line.get('reference') == 'Starting Inventory':
+                        line['quantity'] = line['cumulative_quantity'] if line.get('__count') == 1 else line['quantity']   # Set quantity if line's quantity is 0.0
+                        if index > 1 and line.get('__count') > 1:
+                            line['cumulative_quantity'] = res[index-1]['cumulative_quantity']  # Set cumulatiove quantity if line does not have any cumulative
+                else:
+                    lists.append(line['quantity']) if line.get('quantity') else lists
+                    line['cumulative_quantity'] = sum(lists)  # Sum of all Move Operations' quantity
+                if not line.get('reference'):
```
Comment (amoyaux) [inline addons/stock/report/report_stock_forecast.py]: @hch-odoo are you sure the part after this line is useful? I could translate it by if not groupby and 'reference' not in groupby then line['quantity'] = line['quantity']

**Comment by rdeodoo** [comment]
@amoyaux Looks like the task was set as deployed, done at https://github.com/odoo/odoo/pull/30464/
Can you close this one if not needed anymore?

**Pull request closed without merging**